### PR TITLE
feat: overlapping annotations

### PR DESCRIPTION
**Pull request by jrenee42**
add code to allow the correct tooltip to show up;
plus allow an overlapping point annotation show up within a range annotation

### CODE REVIEW

File: giraffe/src/components/AnnotationLine.tsx [hunk 1]
```diff
@@ -120,7 +120,7 @@ export const AnnotationLine: FunctionComponent<AnnotationLineProps> = props => {
           ${clampedStart - pixelMargin}, ${RANGE_HEIGHT}`,
       fill: color,
       id: props.id,
-      style: {cursor: 'pointer'},
+      style: {cursor: 'pointer', opacity: '60%'},
```
Comment (TCL735) [inline giraffe/src/components/AnnotationLine.tsx:123]: Why 60%? Can we make this into a constant somewhere for easier reference and self documentation?

File: giraffe/src/utils/annotationData.test.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import {sortDistances} from './annotationData'
+
+describe('annotationData utils', () => {
+  describe('sort distances', () => {
+    it('handles empty data', () => {
+      const empty = []
+      empty.sort(sortDistances)
+      expect(empty).toEqual([])
+    })
+    it('a point annotation within the overlap range should be the minimum when there is an overlapping range annotation', () => {
```
Comment (TCL735) [inline giraffe/src/utils/annotationData.test.ts:10]: `it("should read like a sentence", ...)`

`it("does x and y should be the result", ...)`

File: giraffe/src/utils/annotationData.test.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      const points = [
+        {dist: 0, annoType: 'range'},
+        {dist: 2.3, annoType: 'point'},
+      ]
+      const minimum = {dist: 2.3, annoType: 'point'}
+
+      expect(points.sort(sortDistances)[0]).toEqual(minimum)
+    })
+    it('a range and point with same distance, the point should be the minimum', () => {
```
Comment (TCL735) [inline giraffe/src/utils/annotationData.test.ts:19]: Same as above, should read like a sentence.

File: giraffe/src/utils/annotationData.test.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      const points = [
+        {dist: 2.3, annoType: 'range'},
+        {dist: 2.3, annoType: 'point'},
+      ]
+      const minimum = {dist: 2.3, annoType: 'point'}
+      expect(points.sort(sortDistances)[0]).toEqual(minimum)
+    })
+
+    it('a range and point with the range still being closer with the weighted distance', () => {
```
Comment (TCL735) [inline giraffe/src/utils/annotationData.test.ts:28]: Same as above

File: giraffe/src/utils/annotationData.test.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+      const points = [
+        {dist: 0, annoType: 'range'},
+        {dist: 12.3, annoType: 'point'},
+      ]
+      const minimum = {dist: 0, annoType: 'range'}
+      expect(points.sort(sortDistances)[0]).toEqual(minimum)
+    })
+
+    it('two ranges, normal comparison', () => {
```
Comment (TCL735) [inline giraffe/src/utils/annotationData.test.ts:37]: Same as above

File: giraffe/src/utils/annotationData.test.ts [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+      const points = [
+        {dist: 5, annoType: 'range'},
+        {dist: 0, annoType: 'range'},
+      ]
+      const minimum = {dist: 0, annoType: 'range'}
+      expect(points.sort(sortDistances)[0]).toEqual(minimum)
+    })
+
+    it('two points, normal comparison', () => {
```
Comment (TCL735) [inline giraffe/src/utils/annotationData.test.ts:46]: Same as above

File: giraffe/src/utils/annotationData.test.ts (hunk 2 above)
Comment (TCL735) [reply giraffe/src/utils/annotationData.test.ts:10]: The `it` should flow right into the test description like the examples above.

**Review by TCL735 — CHANGES_REQUESTED** [review]

File: giraffe/src/utils/annotationData.test.ts [hunk 7]
```diff
@@ -0,0 +1,55 @@
+import {sortDistances} from './annotationData'
+
+describe('annotationData utils', () => {
+  describe('sort distances', () => {
+    it('handles empty data', () => {
```
Comment (TCL735) [inline giraffe/src/utils/annotationData.test.ts:5]: 👍 great example

File: giraffe/src/components/AnnotationLine.tsx (hunk 1 above)
Comment (jrenee42) [reply giraffe/src/components/AnnotationLine.tsx:123]: that is what looks good; you can see the line while the 100% triangle shows up.  and @daviddkkim  approved it.

Comment (jrenee42) [reply giraffe/src/components/AnnotationLine.tsx:123]: i will add documentation to explain it.

**Review by TCL735 — APPROVED** [review]

**Pull request merged**